writer.go

```diff
@@ -133,6 +134,41 @@ type WriterConfig struct {
 	newPartitionWriter func(partition int, config WriterConfig, stats *writerStats) partitionWriter
 }
 
+type WriterError struct {
+	Msg Message
+	Err error
+}
```

Comment:
Could we have the `WriterError` and `WriterErrors` type implement `errors.Wrapper`?

Reply:
Will do, `Unwrap` is straightforward for `WriterError`. Do you have thoughts on behavior for `WriterErrors`?

Reply:
@achille-roussel For `WriterErrors.Unwrap()` how do you feel about using the same strategy as https://github.com/prometheus/client_golang/blob/82ce871c27ca8919bcdb97d69bcf461499171b27/prometheus/registry.go#L240?